### PR TITLE
yamllinter: raise if a version has patches but no sources

### PR DESCRIPTION
**Pull request by ericLemanissier**
This can happen if "someone" removes an outdated version, but forgets to check if there are patches for this version
(it happened to me yesterday)

---

- [ ] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by ericLemanissier** [comment]
We just found offenders:
- https://github.com/conan-io/conan-center-index/actions/runs/3765974101/jobs/6401984560#step:9:5494 -> https://github.com/conan-io/conan-center-index/blob/8b3c7ac87fccd44d30ae18890823f386d3e12a02/recipes/libcurl/all/conandata.yml#L33
- https://github.com/conan-io/conan-center-index/actions/runs/3765974101/jobs/6401984560#step:9:6276 -> https://github.com/conan-io/conan-center-index/blob/8b3c7ac87fccd44d30ae18890823f386d3e12a02/recipes/libpq/all/conandata.yml#L24
- https://github.com/conan-io/conan-center-index/actions/runs/3765974101/jobs/6401984560#step:9:7538 -> https://github.com/conan-io/conan-center-index/blob/8b3c7ac87fccd44d30ae18890823f386d3e12a02/recipes/mdnsresponder/all/conandata.yml#L18

**Comment by prince-chrismc** [comment]
> * https://github.com/conan-io/conan-center-index/actions/runs/3765974101/jobs/6401984560#step:9:7538 -

This one I remember did weird stuff

https://github.com/conan-io/conan-center-index/blob/8b3c7ac87fccd44d30ae18890823f386d3e12a02/recipes/mdnsresponder/all/conanfile.py#L136

**Comment by ericLemanissier** [comment]
blocked by:
- https://github.com/conan-io/conan-center-index/pull/15499

File: linter/conandata_yaml_linter.py [hunk 1]
```diff
@@ -62,6 +62,14 @@ def main():
     if "patches" in parsed:
         for version in parsed["patches"]:
             patches = parsed["patches"][version]
+            if version not in parsed["sources"]:
+                print(
+                    f"::error file={args.path},line={patches.start_line},endline={patches.end_line},"
```
Comment (prince-chrismc) [inline linter/conandata_yaml_linter.py:67]:
```suggestion
                    f":: warning file={args.path},line={patches.start_line},endline={patches.end_line},"
```

**Review by prince-chrismc — APPROVED** [review]

**Comment by ericLemanissier** [comment]
wait, nothing is reported since I merged master branch into my branch 2 days ago

**Comment by ericLemanissier** [comment]
just to be clear: one of the job failed, but it is not a blocker for this PR

**Review by paulharris — APPROVED** [review]
LGTM

**Review by franramirez688 — APPROVED** [review]

**Pull request merged**